server/debugsession.go

```diff
@@ -0,0 +1,834 @@
+						PresentationHint:   "locals",
+						VariablesReference: int(ScopeIdentifierLocal),
+					},
+					{
+						Name:               "Storage",
+						PresentationHint:   "registers",
+						VariablesReference: int(ScopeIdentifierStorage),
+					},
+				},
+			},
+		})
+
+	case *dap.VariablesRequest:
+		// TODO: reply with error if ds.stop == nil
+		if ds.stop == nil {
+			ds.send(newDAPErrorResponse(request.GetRequest().GetSeq(), "", dap.ErrorMessage{}))
+
+		}
+
+		variableRequested := request.Arguments.VariablesReference
+		responseVariables := make([]dap.Variable, 0)
+
+		inter := ds.stop.Interpreter
+
+		switch ScopeIdentifier(variableRequested) {
+
+		case ScopeIdentifierLocal:
+			//reset variables
+			ds.variableHandleCounter = 0
+			ds.variables = make(map[int]any, 0)
+
+			activation := ds.debugger.CurrentActivation(inter)
+			location := ds.stop.Interpreter.Location
+			functionValues := activation.FunctionValues()
+
+			for name, variable := range functionValues {
+
+				// TODO: generalize exclusion of built-ins / standard library definitions
+
+				if location.String() == ds.scriptID && name == "self" {
+					continue
+				}
+				if name == "BLS" || name == "RLP" {
+					continue
+				}
+
+				value := variable.GetValue()
+
+				cadenceValue, err := runtime.ExportValue(value, inter, interpreter.EmptyLocationRange)
+				if err != nil {
+					//	panic(err)
+					continue
+				}
+				variable := ds.convertValueToDapVariable(name, cadenceValue)
+				responseVariables = append(
+					responseVariables,
+					variable,
+				)
+			}
+
+		case ScopeIdentifierStorage:
+			var index int = 1
+			for {
+				account, err := ds.backend.GetEmulator().GetAccountByIndex(uint(index))
+				if err != nil { //end of accounts
+					break
+				}
+
+				variable := dap.Variable{
+					Name:  account.Address.String(),
+					Value: "FlowAccount",
+					Type:  "FlowAccount",
+					PresentationHint: dap.VariablePresentationHint{
+						Kind:       "class",
+						Visibility: "public",
+					},
+					VariablesReference: ds.storeVariable(account),
+				}
+				index++
+				responseVariables = append(
+					responseVariables,
+					variable,
+				)
+			}
+
+		default:
+			valueRequested := ds.variables[variableRequested]
+			switch value := valueRequested.(type) {
+
+			case *sdk.Account:
+				storage := inter.SharedState.Config.Storage.GetStorageMap(
+					common.Address(value.Address),
+					common.PathDomainStorage.Identifier(),
+					false,
+				)
+				responseVariables = ds.convertStorageMapToDapVariables(inter, storage)
+
+			case interpreter.Value:
+
+			case cadence.Value:
+				responseVariables = ds.convertCadenceValueMembersToDapVariables(value)
+			}
+
+		}
+
+		ds.send(&dap.VariablesResponse{
+			Response: newDAPSuccessResponse(request.GetRequest()),
+			Body: dap.VariablesResponseBody{
+				Variables: responseVariables,
+			},
+		})
+	}
+}
+
+func (ds *debugSession) convertValueToDapVariable(name string, value cadence.Value) dap.Variable {
```

Comment:
```suggestion
func (ds *debugSession) convertValueToDAPVariable(name string, value cadence.Value) dap.Variable {
```

Reply:
Renamed this and related functions in 97ec31a7ec3502b4cad10b5c5822a909224f33ad